frontend/src/metabase/query_builder/components/template_tags/SnippetSidebar.jsx

```diff
@@ -0,0 +1,313 @@
+                    "text-brand bg-light-hover rounded p1 cursor-pointer",
+                  )}
+                  onClick={openSnippetModalWithSelectedText}
+                  name="add"
+                  size={HEADER_ICON_SIZE}
+                />
+                <Icon
+                  className={cx(
+                    { hide: !showSearch },
+                    "text-brand-hover cursor-pointer",
+                  )}
+                  onClick={this.hideSearch}
+                  name="close"
+                  size={HEADER_ICON_SIZE}
+                />
+              </div>
+            </div>
+            {query.databaseId() == null ? (
+              <p className="text-body text-centered">{t`Select a database to see its snippets.`}</p>
+            ) : (
+              <div className="flex-full">
+                {filteredSnippets.map(snippet => (
+                  <SnippetRow
+                    key={snippet.id}
+                    snippet={snippet}
+                    insertSnippet={this.props.insertSnippet}
+                    setModalSnippet={this.props.setModalSnippet}
+                  />
+                ))}
+              </div>
+            )}
+          </div>
+        )}
+      </SidebarContent>
+    );
+  }
+}
+
+@Snippets.loadList({ query: { archived: true }, wrapped: true })
+class ArchivedSnippets extends React.Component {
+  render() {
+    const { onBack, snippets } = this.props;
+    return (
+      <SidebarContent>
+        <SidebarHeader
+          className="p2"
+          title={t`Archived snippets`}
+          onBack={onBack}
+        />
+
+        {snippets.map(snippet => (
+          <SnippetRow
+            key={snippet.id}
+            snippet={snippet}
+            unarchiveSnippet={() => snippet.update({ archived: false })}
+          />
+        ))}
+      </SidebarContent>
+    );
+  }
+}
+class SnippetRow extends React.Component {
+  state: { isOpen: boolean };
+
+  constructor(props) {
+    super(props);
+    this.state = { isOpen: false };
+  }
+
+  render() {
+    const {
+      snippet,
+      insertSnippet,
+      setModalSnippet,
+      unarchiveSnippet,
+    } = this.props;
+    const { description, content } = snippet;
+    const { isOpen } = this.state;
+    return (
+      <div
+        className={cx(
+          { "border-transparent": !isOpen },
+          "border-bottom border-top",
+        )}
+      >
+        <a
+          className="bg-light-hover text-brand-hover flex align-center justify-between p2"
+          onClick={() => this.setState({ isOpen: !isOpen })}
+        >
+          <a
+            className="flex bg-medium-hover rounded hover-parent hover--display text-medium text-brand-hover"
+            style={{ padding: 4 }}
+            onClick={
+              unarchiveSnippet
+                ? () => this.setState({ isOpen: true })
+                : e => {
+                    e.stopPropagation();
+                    insertSnippet(snippet);
```

Comment:
This behavior really surprised me. Maybe we should have a tooltip that says "Insert this snippet" or something?